### PR TITLE
UCM: Put cuda-related code in separate files

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8451/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8456/ for details (Mellanox internal link).

File: src/ucm/cuda/cudamem.c [hunk 1]
```diff
@@ -8,20 +8,328 @@
...
-#include <cuda_runtime.h>
 #include <unistd.h>
 #include <pthread.h>
 
+
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cuMemFree, CUresult,-1, CUdeviceptr)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cuMemFreeHost, CUresult, -1, void *)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cuMemAlloc, CUresult, -1, CUdeviceptr *, size_t)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cuMemAllocManaged, CUresult, -1, CUdeviceptr *,
+                             size_t, unsigned int)
```
Comment (brminich) [inline src/ucm/cuda/cudamem.c]: minor: alignment

File: src/ucm/cuda/cudamem.c [hunk 2]
```diff
@@ -8,20 +8,328 @@
...
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cuMemAllocPitch, CUresult, -1, CUdeviceptr *, size_t *,
+                             size_t, size_t, unsigned int)
```
Comment (brminich) [inline src/ucm/cuda/cudamem.c]: minor: alignment

File: src/ucm/cuda/cudamem.c [hunk 3]
```diff
@@ -8,20 +8,328 @@
...
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cuMemHostGetDevicePointer, CUresult, -1, CUdeviceptr *,
+                             void *, unsigned int)
```
Comment (brminich) [inline src/ucm/cuda/cudamem.c]: minor: alignment

File: src/ucm/cuda/cudamem.c [hunk 4]
```diff
@@ -8,20 +8,328 @@
...
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cuMemHostUnregister, CUresult, -1, void *)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaFree, cudaError_t, -1, void*)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaFreeHost, cudaError_t, -1, void*)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaMalloc, cudaError_t, -1, void**, size_t)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaMallocManaged, cudaError_t, -1, void**, size_t, unsigned int)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaMallocPitch, cudaError_t, -1, void**, size_t *,
+                             size_t, size_t)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaHostGetDevicePointer, cudaError_t, -1, void**,
+                             void *, unsigned int)
```
Comment (brminich) [inline src/ucm/cuda/cudamem.c]: minor: alignment

File: src/ucm/cuda/cudamem.c [hunk 5]
```diff
@@ -8,20 +8,328 @@
...
+                             size_t, size_t, unsigned int)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cuMemHostGetDevicePointer, CUresult, -1, CUdeviceptr *,
+                             void *, unsigned int)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cuMemHostUnregister, CUresult, -1, void *)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaFree, cudaError_t, -1, void*)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaFreeHost, cudaError_t, -1, void*)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaMalloc, cudaError_t, -1, void**, size_t)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaMallocManaged, cudaError_t, -1, void**, size_t, unsigned int)
+UCM_DEFINE_REPLACE_DLSYM_FUNC(cudaMallocPitch, cudaError_t, -1, void**, size_t *,
+                             size_t, size_t)
```
Comment (brminich) [inline src/ucm/cuda/cudamem.c]: minor: alignment

File: src/ucm/cuda/cudamem.c [hunk 6]
```diff
@@ -40,23 +348,31 @@ static ucm_reloc_patch_t patches[] = {
...
-ucs_status_t ucm_cudamem_install()
+static ucs_status_t ucm_cudamem_install(int events)
 {
     static int ucm_cudamem_installed = 0;
     static pthread_mutex_t install_mutex = PTHREAD_MUTEX_INITIALIZER;
     ucm_reloc_patch_t *patch;
     ucs_status_t status;
 
+    if (!(events & (UCM_EVENT_MEM_TYPE_ALLOC | UCM_EVENT_MEM_TYPE_FREE))) {
+        status = UCS_OK;
```
Comment (brminich) [inline src/ucm/cuda/cudamem.c]: minor: can init `status` to `UCS_OK` at initialization phase. Then would not need to set it to `UCS_OK` three times

**Comment by yosefe** [comment]
@brminich fixed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8472/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8477/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@bureddy can you pls take a look?

**Pull request merged**